test/index.js

```diff
@@ -105,6 +105,22 @@ test('test a react component with custom name function', function(t) {
   t.equal(output, '<Basic test1={_testCallBack1}\n  test2={_testCallBack2} />');
 });
 
+test('test a react component with autodetection of the functions names', function(t) {
+  t.plan(1);
+
+  let _testCallBack1 = function () {
+    //no-op
+  };
+
+  let output = jsxToString(
+    <Basic test1={_testCallBack1} test2={function() {}} />, {
```

Comment:
`why an anonymous function is being converted to test2 name?`
I think this is how the `name` property works for the anonymous functions, that are attached to a property. For example, if you assign an anonymous function to a variable, the `name` property will be the variable's name. 

`if there is a way to provide the implementation for the function`
This is exactly what I'm trying to do :) I'm writing a plugin for react storybook (https://github.com/storybooks/react-storybook), which shows the content of a story in a side panel. This panel will be divided in two, JSX on top and related js at the bottom. It's going to look like this:
![screen shot 2016-12-09 at 8 58 26 am](https://cloud.githubusercontent.com/assets/6586830/21029423/bc7fb656-bded-11e6-866d-bcaa9763b318.png)

So having functions' names in there without the need of manually overwriting all of the properties with functions would be super helpful.